qiskit/aqua/algorithms/minimum_eigen_solvers/vqe.py

```diff
@@ -218,13 +218,18 @@ def aux_operators(self,
         self._aux_op_nones = None
         if isinstance(aux_operators, list):
             self._aux_op_nones = [op is None for op in aux_operators]
-            zero_op = I.tensorpower(self.operator.num_qubits) * 0.0
-            converted = [op.to_opflow() if op else zero_op for op in aux_operators]
+            converted = [op.to_opflow() if isinstance(op, LegacyBaseOperator)
+                         else op for op in aux_operators]
```

Comment:
I think you removed some functionality here;  if the `aux_operators` contains `None` then a `zero_op` should be inserted, now the `None` remains. Functionally you would need to do something like
```python
converted = []
for op in aux_operators:
    if op is None:
        converted += [zero_op]
    elif isinstance(op, LegacyBaseOperator):
        converted += [op.to_opflow()]
    else:
        converted += [op]
```

Reply:
I did, intending to keep ``None`` elements in the list as such. Based on the use of ``_aux_op_nones`` in 
```python
if isinstance(aux_operators, list):
            self._aux_op_nones = [op is None for op in aux_operators]
```
and the following in ``_eval_aux_ops``,
```python
# Deal with the aux_op behavior where there can be Nones or Zero qubit Paulis in the list
        self._ret['aux_ops'] = [None if is_none else [result]
                                for (is_none, result) in zip(self._aux_op_nones, aux_op_results)]
        self._ret['aux_ops'] = np.array([self._ret['aux_ops']])
```
it seemed that the intended output in ``_ret['aux_ops']`` is ``None`` for operators that are ``None``, and 0 for operators that are zero-qubit operators assigned the value of ``zero_op``. Keeping ``None`` operators explicitly is also consistent with the behavior of the setter in NumPyEigensolver.
I must add that I personally don't see a use case for having either zero-qubit or None operators.

Reply:
The aux_operators are passed in with the understanding that a list of values, in the same order will be passed back. Each value in that list resulting from the aux operator being measured in the same state that the main operator value came from. This was originally done so in chemistry the stack could emit other operators like dipole, spin and particle number and have these measured too. Since we did not want to make VQE domain dependent these were called aux_operators and VQE has no understanding of what they are exactly. So they can be any other operators a user wants to pass. A user can assemble them in an order it knows and pass them in knowing it will get the results back in the same order.
So out of chem stack comes a main hamiltonian and an optional list of aux_operators. Often these would be put directly into VQE. Then we added Z2 symmetry reduction which allows the main hamiltonian to be split by any symmetries into smaller problems - one of which will have the ground state. So a user taking the output from chemistry and manipulating the operators can find, if there are symmetries, the problem can be solved using less qubits. Thing is the aux_operators have to be acted upon the same way by the symmetries but if they do not commute with that then the measurement is meaningless. Hence we allowed them to be be dropped from the list that would go to VQE etc and None be put in its place - so the overall index of where they are was still maintained.  These can all then be given to VQE. Then when the original creator, say the chemistry stack, is given the result from VQE to interpret it, it should expect to get back None though as a value in the returned value list. The order is still the same as it originally gave so it know what aux operator each value was from. This is how we ended up allowing None to allow an intermediary between the original creator of the list and VQE etc to change the aux_operators but where that change might make them meaningless to evaluate.
Before the opflow we simply skipped over None internally in VQE etc placing None in the output value.

Reply:
> Then when the original creator, say the chemistry stack, is given the result from VQE to interpret it, it should expect to get back None though as a value in the returned value list. The order is still the same as it originally gave so it know what aux operator each value was from. This is how we ended up allowing None to allow an intermediary between the original creator of the list and VQE etc to change the aux_operators but where that change might make them meaningless to evaluate.

I see, thank you for the context! In that case, I believe my change that keeps ``None`` values as such and does not convert them to zero restores the original behavior before opflow.

Reply:
The other aspect of the aux_operators that it is possible for an operator in the list to be empty. In the past we made an explicit is_empty check and set the return value to 0 bypassing the normal logic to compute the expectation value. So thus None and empty operators may end up needing special treatment.
I guess its true that we did not do any upfront check on the operator compatibility in regards of number of qubits. I think in this case though, if such a check is added, it is best to raise an error so it can be corrected rather than ignoring it by dropping the operator inside VQE.